src/components/TxHistory.tsx

```diff
@@ -0,0 +1,338 @@
+import { keyframes, styled } from 'styled-components';
+import { Theme, WhiteLabelTheme } from '../theme.types';
+import { useServiceContext } from '../hooks/useServiceContext';
+import { useEffect, useMemo, useState } from 'react';
+import { HeaderText, Text } from './Reusable';
+import { GENERIC_TOKEN_ICON } from '../utils/constants';
+import { FaTimes, FaChevronDown, FaChevronUp } from 'react-icons/fa';
+import { TxLog } from 'spv-store';
+import transactions from './const';
+import { FaExternalLinkAlt } from 'react-icons/fa';
+import { Button } from './Button';
+import bsvCoin from '../assets/bsv-coin.svg';
+import lock from '../assets/lock.svg';
+import priceTag from '../assets/price-tag.svg';
+
+const slideIn = keyframes`
+  from {
+    transform: translateY(100%);
+  }
+  to {
+    transform: translateY(0);
+  }
+`;
+
+const slideOut = keyframes`
+  from {
+    transform: translateY(0);
+  }
+  to {
+    transform: translateY(100%);
+  }
+`;
+
+const Container = styled.div<{ isSlidingOut: boolean } & WhiteLabelTheme>`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  width: 100%;
+  height: 100vh;
+  overflow-y: auto;
+  background-color: ${({ theme }) => theme.color.global.walletBackground};
+  z-index: 1000;
+  position: absolute;
+  animation: ${({ isSlidingOut }) => (isSlidingOut ? slideOut : slideIn)} 0.75s forwards;
+`;
+
+const FavoriteRow = styled.div<WhiteLabelTheme>`
```

Comment:
Rename to TxRow or HistoryRow or something that makes more sense for this TxHistory component.